pkg/interface/src/components/Omnibox.js

```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import { Box, Row, Rule, Text } from '@tlon/indigo-react';
+import index from '../lib/omnibox';
+import Mousetrap from 'mousetrap';
+import OmniboxResult from './OmniboxResult';
+
+import { cite } from '../lib/util';
+
+export class Omnibox extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: new Map([]),
+      query: '',
+      results: this.initialResults(),
+      selected: ''
+    };
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.search = this.search.bind(this);
+    this.navigate = this.navigate.bind(this);
+    this.control - this.control.bind(this);
+    this.setPreviousSelected = this.setPreviousSelected.bind(this);
+    this.setNextSelected = this.setNextSelected.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
+      this.setState({ index: index(this.props.associations, this.props.apps.tiles) });
+    }
+
+    if (prevProps && this.props.show && prevProps.show !== this.props.show) {
+      Mousetrap.bind('escape', () => this.props.api.local.setOmnibox());
+      document.addEventListener('mousedown', this.handleClickOutside);
+      this.input.focus();
+    }
+  }
+
+  componentWillUpdate(prevProps) {
+    if (!this.props.show && prevProps.show !== this.props.show) {
+      Mousetrap.unbind('escape');
+      document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+  }
+
+    control(evt) {
+    if (evt.key === 'Escape') {
+      if (this.state.query.length > 0) {
+        this.setState({ query: '', results: this.initialResults() });
+      } else if (this.props.show) {
+        this.props.api.local.setOmnibox();
+      }
+    };
+
+    if (evt.key === 'ArrowDown') {
+      this.setNextSelected();
+    }
+
+    if (evt.key === 'ArrowUp') {
+      this.setPreviousSelected();
+    }
+
+    if (evt.key === 'Enter') {
+      if (this.state.selected !== '') {
+        this.navigate(this.state.selected);
+      } else {
+        this.navigate(Array.from(this.state.results.values()).flat()[0].link);
+      }
+    }
+  }
+
+  handleClickOutside(evt) {
+    if (this.props.show && !this.omniBox.contains(evt.target)) {
+      this.props.api.local.setOmnibox();
+    }
+  }
+
+  initialResults() {
+    return new Map([
+      ['commands', []],
+      ['subscriptions', []],
+      ['groups', []],
+      ['apps', []]
+    ]);
+  }
+
+  navigate(link) {
+    const { props } = this;
+    this.setState({ results: this.initialResults(), query: '' }, () => {
+      props.api.local.setOmnibox();
+      props.history.push(link);
+    });
+  }
+
+  search(event) {
+    const { state } = this;
+    const query = event.target.value;
+    const results = this.initialResults();
+
+    this.setState({ query: query });
+
+    // wipe results if backspacing
+    if (query.length === 0) {
+      this.setState({ results: results });
```

Comment:
this should also reset `state.selected`, otherwise you have jank where it selects the most recent selection, after you've cleared it.